azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -89,12 +89,14 @@ def poll_all_tracked():
                         logger.periodic_warn(logger.EVERY_HOUR, '[PERIODIC] Could not collect metrics for cgroup '
                                                                 '{0}. Error : {1}'.format(cgroup.name, ustr(e)))
                 if not cgroup.is_active():
-                    CGroupsTelemetry.stop_tracking(cgroup)
+                    inactive_cgroups.append(cgroup)
```

Comment:
Any particular reason for adding the inactive cgroups to an array first and then removing them later? Like why not just `stop_tracking` them here itself?

Reply:
Of course I tried like you mentioned and learned this runtime error. So I came up with this array workaround. (_RuntimeError: dictionary changed size during iteration_)